src/backend/config.ts

```diff
@@ -1,37 +1,47 @@
-export function checkVariables() {
-  [
+export function checkVariables(
+  envVars: NodeJS.ProcessEnv,
+  requiredKeys: Array<keyof NodeJS.ProcessEnv>,
+) {
+  const missingKeys = requiredKeys.filter((key) => !envVars[key]);
+  if (missingKeys.length > 0) {
+    throw `Variable(s) ${missingKeys.join(', ')} are missing from your environment`;
+  }
+}
+
+export function getConfig() {
+  checkVariables(process.env, [
     'ENVIRONMENT',
     'PORT',
+    'DB_NAME',
+    'DB_USERNAME',
+    'DB_PASSWORD',
+    'DB_HOST',
+    'DB_PORT',
     'DISCOURSE_URL',
     'DISCOURSE_SSO_SECRET',
     'COOKIE_SECRET',
     'JWT_SECRET',
     'LETTER_ACCESS_KEY_SALT',
     'LETTER_AES_KEY',
-    'DB_NAME',
-    'DB_USERNAME',
-    'DB_PASSWORD',
-    'DB_HOST',
-    'DB_PORT',
-  ].forEach((key) => {
-    if (!(key in process.env)) {
-      throw `Missing variable ${key} from your environment`;
-    }
-  });
-}
+  ]);
 
-export function getConfig() {
-  checkVariables();
+  // `checkVariables` should ensure required variables are available here. Unfortunately typescript
+  // is not able to pick this up automatically. So disabling no-non-null-assertion eslint rule here.
+  /* eslint-disable @typescript-eslint/no-non-null-assertion */
+  const hostname = process.env.APP_DOMAIN || 'localhost';
```

Comment:
Use `APP_DOMAIN` instead of `HOSTNAME` because in some operating systems, `HOSTNAME` is defined globally and picked up automatically by the node process.